### PR TITLE
Additional vector tests

### PR DESCRIPTION
**Pull request by nkorinek**
Have to merge https://github.com/earthlab/matplotcheck/pull/203 before this one.

### CODE REVIEW

**Comment by mbjoseph** [comment]
Heya @nkorinek - not 100% sure whether this is a fix, but do you need to set the matplotlib backend to get these commented out tests to work? e.g., in `test_vector.py` at the top:

```python
... # other imports
import matplotlib
matplotlib.use('Agg')

# awesome tests
...
```

**Comment by nkorinek** [comment]
@lwasser Alright I think I'm happy with how much of the code is tested now! Let me know what you think and if I need to look into adding anymore vector tests.

**Comment by lwasser** [comment]
> Heya @nkorinek - not 100% sure whether this is a fix, but do you need to set the matplotlib backend to get these commented out tests to work? e.g., in `test_vector.py` at the top:
> 
> ```python
> ... # other imports
> import matplotlib
> matplotlib.use('Agg')
> 
> # awesome tests
> ...
> ```

THANK YOU @mbjoseph !! so appreciative of your help!

File: matplotcheck/tests/conftest.py [hunk 1]
```diff
@@ -159,12 +159,26 @@ def pt_time_line_plt(pd_df_timeseries):
...
-    ax.set_ylabel("y label")
+    for ctype, points in pd_gdf.groupby("attr"):
+        color = point_symb[ctype]
+        label = ctype
+        size += 100
+        points.plot(color=color, ax=ax, label=label, markersize=size)
+
+    ax.legend(title="Legend", loc=(1.1, 0.1))
 
     axis = plt.gca()
```
Comment (lwasser) [inline matplotcheck/tests/conftest.py]: @nkorinek let's move away from using plt.gca. just call PlotTester(ax)

File: matplotcheck/tests/test_lines.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+@pytest.fixture
+def mixed_type_geo_plot(pd_gdf, multi_line_gdf):
+    """Create a point plot for testing"""
+    _, ax = plt.subplots()
+
+    pd_gdf.plot(ax=ax)
+    multi_line_gdf.plot(ax=ax)
+
+    axis = plt.gca()
```
Comment (lwasser) [inline matplotcheck/tests/test_lines.py]: here as well - avoid plt.gca and call ax instead to create your tester object. please check for this throughout

File: matplotcheck/tests/test_points.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+    for ctype, points in pd_gdf.groupby("attr"):
+        color = point_symb[ctype]
+        label = ctype
+        size += 100
+        points.plot(color=color, ax=ax, label=label, markersize=size)
+
+    ax.legend(title="Legend", loc=(1.1, 0.1))
+
+    axis = plt.gca()
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: i'm not commenting on each time this is implemented but you get the idea. no more plt.gca!

File: matplotcheck/tests/test_points.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    pd_gdf.plot(ax=ax)
+
+    axis = plt.gca()
+
+    return VectorTester(axis)
+
+
+@pytest.fixture
+def pt_geo_plot_edge(edge_gdf, two_line_gdf):
+    """Create a point plot for edge case testing"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: let's clarify what these fixtures are. this is a point plot grouped by attribute. we know grouped features do have specific characteristics. 

Let's also consider more specific expressive names for our fixtures. i'm not sure what geo_plot_edge means. what is the edge case? can you come up with better names please? i'm happy to work with you on this.

File: matplotcheck/tests/test_points.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    two_line_gdf.plot(ax=ax)
+
+    ax.legend(title="Legend", loc=(1.1, 0.1))
+
+    axis = plt.gca()
+
+    return VectorTester(axis)
+
+
+def test_points_sorted_by_markersize_pass(pt_geo_plot, pd_gdf):
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py:76]: this is a nice expressive name!

File: matplotcheck/tests/test_points.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    """Test points sorted by size of attribute pass"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: do you mean attribute value rather than size of attribute?

File: matplotcheck/tests/test_points.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    pt_geo_plot.assert_collection_sorted_by_markersize(pd_gdf, "attr")
+    plt.close("all")
+
+
+def test_points_sorted_by_markersize_fail(pt_geo_plot_bad, pd_gdf):
+    """Test points sorted by size of attribute fails"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: attribute value? and also do you mean sorted or grouped?

File: matplotcheck/tests/test_points.py [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+def test_assert_points_custom_message(pt_geo_plot, bad_pd_gdf):
+    """Tests that a custom error message is passed."""
+    message = "Test message"
+    with pytest.raises(AssertionError, match="Test message"):
+        pt_geo_plot.assert_points(points_expected=bad_pd_gdf, m=message)
+        plt.close("all")
+
+
+def test_wrong_length_points_expected(pt_geo_plot, pd_gdf, bad_pd_gdf):
+    """Tests that error is thrown for incorrect lenght of a gdf"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]:
```suggestion
    """Tests that error is thrown for incorrect length of a gdf"""
```

Comment (lwasser) [reply matplotcheck/tests/test_points.py]: @nkorinek i'm not sure what this means. gdfs don't have lengths? do you mean number of rows perhaps?

File: matplotcheck/tests/test_points.py [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+    with pytest.raises(AssertionError, match="points_expected's length does "):
+        pt_geo_plot.assert_points(bad_pd_gdf.append(pd_gdf), "attr")
+        plt.close("all")
+
+
+def test_convert_length_error(pt_geo_plot):
+    """Test that the convert lenght function throws an error"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: let's talk about convert length. i'm just not sure what this means! also length is spelled incorrectly :)

File: matplotcheck/tests/test_points.py [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    with pytest.raises(ValueError, match="Input array length is not: 1 or 9"):
+        pt_geo_plot._convert_length(np.array([1, 2, 3, 4]), 9)
+
+
+def test_point_gdf_with_zeros(pt_geo_plot_edge, edge_gdf):
+    """Test that assert_points works when there's a zero point in the gdf"""
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py]: what is a zero point?

File: matplotcheck/tests/test_polygons.py [hunk 11]
```diff
@@ -2,82 +2,124 @@
 import pytest
 import matplotlib.pyplot as plt
 import geopandas as gpd
+from shapely.geometry import Polygon
 from matplotcheck.vector import VectorTester
+import matplotlib
```
Comment (lwasser) [inline matplotcheck/tests/test_polygons.py]: please use pep8 throughout. matplotlib should come before matplotcheck and geopandas! it should be at the top in this case.

File: matplotcheck/tests/test_polygons.py [hunk 12]
```diff
@@ -2,82 +2,124 @@
...
+    """Create a line vector tester object."""
     _, ax = plt.subplots()
 
-    basic_polygon_gdf.plot(ax=ax)
-    ax.set_title("My Plot Title", fontsize=30)
-    ax.set_xlabel("x label")
-    ax.set_ylabel("y label")
+    two_line_gdf.plot(ax=ax)
 
     axis = plt.gca()
```
Comment (lwasser) [inline matplotcheck/tests/test_polygons.py]: more plt.gca()

File: matplotcheck/tests/test_polygons.py [hunk 13]
```diff
@@ -2,82 +2,124 @@
...
 
     return VectorTester(axis)
 
 
 @pytest.fixture
-def point_geo_plot(pd_gdf):
-    """Create a point plot for testing"""
+def poly_multiline_plot(multi_line_gdf):
```
Comment (lwasser) [inline matplotcheck/tests/test_polygons.py]: just curious. this says poly multiline? is it a multiline object or a polygon? please clarify

File: matplotcheck/tests/test_polygons.py [hunk 14]
```diff
@@ -2,82 +2,124 @@
...
 
 @pytest.fixture
-def bad_pd_gdf(pd_gdf):
-    """Create a point geodataframe with slightly wrong values for testing"""
-    return gpd.GeoDataFrame(
-        geometry=gpd.points_from_xy(
-            pd_gdf.geometry.x + 1, pd_gdf.geometry.y + 1
-        )
-    )
+def poly_multiline_plot_bad(multi_line_gdf):
```
Comment (lwasser) [inline matplotcheck/tests/test_polygons.py]: same comment here. polygon or line object??

File: matplotcheck/tests/test_polygons.py [hunk 15]
```diff
@@ -109,39 +151,10 @@ def test_polygon_custom_fail_message(poly_geo_plot, basic_polygon):
...
-    with pytest.raises(ValueError, match="points_expected is not expected"):
-        point_geo_plot.assert_points(points_expected=list_geo)
-
-
-def test_get_points(point_geo_plot, pd_gdf):
-    """Tests that get_points returns correct values"""
-    xy_values = point_geo_plot.get_points()
-    assert list(sorted(xy_values.x)) == sorted(list(pd_gdf.geometry.x))
-    assert list(sorted(xy_values.y)) == sorted(list(pd_gdf.geometry.y))
+        plt.close("all")
```
Comment (lwasser) [inline matplotcheck/tests/test_polygons.py:115]: just curious -- did we move the point tests somewhere else? just want to ensure i understand this change and it's documented!

**Review by lwasser — CHANGES_REQUESTED** [review]
ok @nkorinek this is looking good! just some more cleanup. i'm noticing there are a lot of tests with names that don't describe the test or fixture well. let's work on cleaning up this pr as well with better test and fixture names, removing plt.gca() etc!

File: matplotcheck/tests/test_points.py (hunk 9 above)
Comment (nkorinek) [reply matplotcheck/tests/test_points.py]: Hey! So, this test was making sure that the helper function, `_convert_length`, which is used to help the `get_points_by_attributes` and `get_lines_by_attributes`, fails when given incorrect inputs. I was really just making sure it threw the correct error, and wasn't sure how to get it to throw that error aside from giving it direct inputs.

File: matplotcheck/tests/test_points.py (hunk 10 above)
Comment (nkorinek) [reply matplotcheck/tests/test_points.py]: It's a point at origin. There was a specific bug around origin points that I ironed out, but this makes sure that code works. I've re-worded the names of the geodataframes, test objects, and tests themselves to clarify what this does.

File: matplotcheck/tests/test_polygons.py (hunk 13 above)
Comment (nkorinek) [reply matplotcheck/tests/test_polygons.py]: Sorry, this is a bad name. I'm going to rename it `multiline_geo_plot` to more closely match the point and polygon plots.

File: matplotcheck/tests/test_polygons.py (hunk 15 above)
Comment (nkorinek) [reply matplotcheck/tests/test_polygons.py:115]: Yep! They were moved to the `test_points.py` file. If you look all the tests that were removed from here exist over there.

**Comment by nkorinek** [comment]
Ok @lwasser , I went through and renamed variables you pointed out had confusing names, and hopefully clarified a lot of the confusion. I think this is ready for another review!

File: CHANGELOG.md [hunk 16]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+* Created test for the vector module (@nkorinek, #209)
```
Comment (lwasser) [inline CHANGELOG.md]:
```suggestion
* Created tests for the vector module (@nkorinek, #209)
```

File: matplotcheck/tests/test_polygons.py (hunk 15 above)
Comment (lwasser) [reply matplotcheck/tests/test_polygons.py:115]: ok thank you for the clarification!

File: matplotcheck/tests/test_points.py [hunk 17]
```diff
@@ -0,0 +1,161 @@
+"""Tests for the vector module"""
+import numpy as np
+import matplotlib
+import pytest
+import matplotlib.pyplot as plt
```
Comment (lwasser) [inline matplotcheck/tests/test_points.py:4]: @nkorinek please move the matplotlib import under the abve matplotlib import to follow PEP8

Comment (lwasser) [reply matplotcheck/tests/test_points.py:4]: i just fixed it. super minor!

**Comment by lwasser** [comment]
ok once CI finishes I will merge this!!

**Review by lwasser — APPROVED** [review]
